### PR TITLE
Improves implant tracking by using a list instead of world

### PR DESCRIPTION
**Pull request by Jalleo**
This is a shameless rip of https://github.com/d3athrow/vgstation13/pull/5691
If it doesn't work blame @Clusterfack

### CODE REVIEW

File: code/game/objects/items/weapons/implants/implant.dm [hunk 1]
```diff
@@ -87,6 +87,18 @@
...
+/obj/item/weapon/implant/tracking/New()
+	..()
+	tracking_implants += src
+
+/obj/item/weapon/implant/tracking/Destroy()
+	..()
+	tracking_implants -= src
+
+
+var/global/tracking_implants = list()
```
Comment (RemieRichards) [inline code/game/objects/items/weapons/implants/implant.dm]: ```var/global/list/tracking_implants = list()```, never leave a list as just a ```var/```

Comment (Miauw62) [inline code/game/objects/items/weapons/implants/implant.dm]: if you define it outside of anything the /global/ isnt necessary, this should be ```var/list/tracking_implants = list()```, so you can do things like tracking_implants.len.

Comment (Jalleo) [inline code/game/objects/items/weapons/implants/implant.dm]: Done

Comment (RemieRichards) [inline code/game/objects/items/weapons/implants/implant.dm]: @Miauw62 YOU FUCK.
@Jalleo put the /global back.

```var/global/list/global_tracking_implants = list()```, and move it to the code/_globalvars/lists.dm file.

**Comment by Miauw62** [comment]
wait why is the /global/ neccesary am i dumb?

**Comment by RemieRichards** [comment]
@Miauw62 if it's just ```/list```, files above it can't interact with it, which is just one less thing to worry about if you put ```/global``` on it.

**Comment by Jalleo** [comment]
Welp lemme get global on it

**Comment by RemieRichards** [comment]
@Jalleo put it in code/_globalvars/lists.dm

**Comment by Jalleo** [comment]
Dammit that timing was just out

**Comment by Core0verload** [comment]
We have other remotely controlled implants. All implamts list would be better than a list for tracking implants only.

**Comment by Jalleo** [comment]
@CoreOverload ill do that tomorrow by working out which should be in what list to make it as efficent as possible. While all in one list may be easier having them in seperate ones reduces runtime delay.

**Comment by Core0verload** [comment]
Even 500 items list is a lot faster than word. But if you want to be efficent, just add implants to list on Implant() and remove them from list on Remove().

**Comment by Jalleo** [comment]
Alright found a whole bunch of other in worlds that is tracking the tracking implants/chem implants no other implants that I know of are being tracked apart from these two.

File: code/_globalvars/lists/objects.dm [hunk 2]
```diff
@@ -14,4 +14,5 @@ var/global/list/chemical_reagents_list				//list of all /datum/reagent datums in
 var/global/list/surgeries_list = list()				//list of all surgeries by name, associated with their path.
 var/global/list/table_recipes = list()				//list of all table craft recipes
 var/global/list/rcd_list = list()					//list of Rapid Construction Devices.
-var/global/list/apcs_list = list()					//list of all Area Power Controller machines, seperate from machines for powernet speeeeeeed.
\ No newline at end of file
+var/global/list/apcs_list = list()					//list of all Area Power Controller machines, seperate from machines for powernet speeeeeeed.
+var/global/list/tracking_implants = list()			//list of all tracking implants to work out what treks everyone are on. Sadly not on lavaworld not implemented...
```
Comment (RemieRichards) [inline code/_globalvars/lists/objects.dm]: this really needs renaming to ```implants_list``` as it now contains more than tracking ones

Comment (Jalleo) [inline code/_globalvars/lists/objects.dm]: technically it should be tracked_implants since chem implants are tracked

Comment (RemieRichards) [inline code/_globalvars/lists/objects.dm]: I supposed, the point is the name is now deceptive.

Comment (Jalleo) [inline code/_globalvars/lists/objects.dm]: ill do a fix PR gonna move tracking implants out of the general implant file to make it clearer anyhow.

**Pull request merged**